Fluid/FluidTemplateExtensions.cs

```diff
@@ -44,45 +44,25 @@ public static ValueTask<string> RenderAsync(this IFluidTemplate template, Templa
                 ExceptionHelper.ThrowArgumentNullException(nameof(template));
             }
 
-            static async ValueTask<string> Awaited(
-                ValueTask task,
-                StringWriter writer,
-                StringBuilderPool builder)
-            {
-                await task;
-                await writer.FlushAsync();
-                var s = builder.ToString();
-                builder.Dispose();
-                writer.Dispose();
-                return s;
-            }
-
             var sb = StringBuilderPool.GetInstance();
             var writer = new StringWriter(sb.Builder);
 
+            // A template is evaluated in a child scope such that the provided TemplateContext is immutable
+            context.EnterChildScope();
+
             try
             {
```

Comment:
What will be returns if an exception is occured?!!

Reply:
And Exception

Reply:
So this intentionally?!! coz isn't throw before

Reply:
Probably we need a remark on that method to make clear to everyone

Reply:
I don't think the behavior has changed here. New code just awaits to prevent control flow from getting out with unfinished continuations that could run after finally has been processed.

Reply:
What confuses me the old behavior returns the actual string from the ` StringBuilderPool` instace in both case (Exception or Without Exception) but now the string is returned if no exception occurs. Am I right?

Reply:
The old code would have done the `await` on `Awaited` method which would have bubbled the exception. So both versions will throw, this one obeys template context life time.

Reply:
Wrong, if there had been an exception, it would have been thrown by the task that is returned.
No exception is caught here, so everything will bubble up.